src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml

```diff
@@ -2387,7 +2411,90 @@ components:
       type: string
       enum:
         - ResourceBadge
-        - NonFungibleResourceBadge
+        - NonFungibleBadge
+
+    AccountDefaultDepositRule:
+      type: string
+      description: Default deposit rule set to an account.
+      enum:
+        - Accept
+        - Reject
+        - AllowExisting
+
+    AccountDepositPreValidationDecidingFactors:
+      description: Deciding factors used to calculate response.
+      type: object
+      required:
+        - default_deposit_rule
+      properties:
+        is_badge_authorized_depositor:
+          description: Whether the input badge belongs to the account's set of authorized depositors. This field will only be present if any badge was passed in the request.
+          type: boolean
+          nullable: true
+        default_deposit_rule:
+          $ref: "#/components/schemas/AccountDefaultDepositRule"
+        resource_specific_details:
+          description: Returns deciding factors for each resource. Contains only information about resources presented in the request, not all resource preference rules for queried account.
+          type: array
+          items:
+            $ref: "#/components/schemas/AccountDepositPreValidationDecidingFactorsResourceSpecificDetailsItem"
+
+    AccountDepositPreValidationDecidingFactorsResourceSpecificDetailsItem:
+      type: object
+      required:
+        - resource_address
+        - vault_exists
+        - is_xrd
+      properties:
+        resource_address:
+          $ref: "#/components/schemas/Address"
+        vault_exists:
+          type: boolean
+        is_xrd:
+          type: boolean
+        resource_preference_rule:
+          $ref: "#/components/schemas/AccountResourcePreferenceRule"
+
+    AccountDepositPreValidationResourceSpecificBehaviourItem:
+      type: object
+      required:
+        - resource_address
+        - allows_try_deposit
+      properties:
+        resource_address:
+          $ref: "#/components/schemas/Address"
+        allows_try_deposit:
+          type: boolean
+
+    TransactionAccountDepositPreValidationAuthorizedDepositorBadge:
+      type: object
+      required:
+        - badge_type
+        - resource_address
+      properties:
+        badge_type:
+          $ref: "#/components/schemas/AccountAuthorizedDepositorBadgeType"
+        resource_address:
```

Comment:
I believe we were supposed to move `resource_address` to derived types as the fact that it exists in all derived types today is nothing more but a coincidence.

Reply:
yes, but couldn't force openAPI generator to properly generate classes if that was inside child classes.